cherab/tools/observers/bolometry.py

```diff
@@ -687,6 +695,350 @@ def etendue_single_run(_):
+      with multiple calls to observe. Defaults to False.
+    :param float curvature_radius: Detectors in real bolometer cameras may
+      have curved corners due to machining limitations. This parameter species
+      the corner radius.
+
+    :ivar Vector3D normal_vector: The normal vector of the detector constructed from
+      the cross product of the x and y basis vectors.
+    :ivar float width: The extent of the bolometer foil in the basis_x direction.
+    :ivar float height: The extent of the bolometer foil in the basis_y direction.
+    :ivar array pixels_as_foils: A 2D array of pixels as individual BolometerFoil objects,
+      useful for calling BolometerFoil methods on each pixel (e.g. sightline tracing).
+      The array is indexed by (column, row).
+    :ivar Vector3D sightline_vectors: A 2D array of vectors that point from the centre
+      of each pixel on the foil detector to the centre of the slit. Defines the
+      effective sightline vectors of the pixels of the detector. The list is indexed by
+      (column, row).
+
+    .. code-block:: pycon
+
+       >>> from raysect.core import Point3D, Vector3D
+       >>> from raysect.optical import World
+       >>> from cherab.tools.observers import BolometerIRVB
+       >>>
+       >>> world = World()
+       >>>
+       >>> # construct basis vectors
+       >>> basis_x = Vector3D(1, 0, 0)
+       >>> basis_y = Vector3D(0, 1, 0)
+       >>> basis_z = Vector3D(0, 0, 1)
+       >>>
+       >>> # specify a detector, you need already created slit and camera objects
+       >>> width = 0.0025
+       >>> pixels = (10, 20)
+       >>> centre_point = Point3D(0, 0, -0.08)
+       >>> detector = BolometerIRVB("ch#1", centre_point, basis_x, basis_y, width, pixels, slit, parent=camera)
+    """
+
+    def __init__(self, detector_id, centre_point, basis_x, basis_y, width, pixels, slit,
+                 parent=None, units="Power", accumulate=False, curvature_radius=0):
+
+        # perform validation of input parameters
+
+        if not isinstance(width, (float, int)):
+            raise TypeError("width argument for BolometerIRVB must be of type float/int.")
+        if width < 0:
+            raise ValueError("width argument for BolometerIRVB must be greater than zero.")
+
+        if not isinstance(slit, BolometerSlit):
+            raise TypeError("slit argument for BolometerIRVB must be of type BolometerSlit.")
+
+        if not isinstance(centre_point, Point3D):
+            raise TypeError("centre_point argument for BolometerIRVB must be of type Point3D.")
+
+        if not isinstance(curvature_radius, (float, int)):
+            raise TypeError("curvature_radius argument for BolometerIRVB "
+                            "must be of type float/int.")
+        if curvature_radius < 0:
+            raise ValueError("curvature_radius argument for BolometerIRVB "
+                             "must not be negative.")
+
+        if not isinstance(basis_x, Vector3D):
```

Comment:
Explicitly convert to Vector3D, then test properties. The user should then be able to supply e.g. a tuple to the init: (1.0, 0.0, 0.0):

```
basis_x = Vector3D(*basis_x)
```

Reply:
Done.